base/iobuffer.jl

```diff
@@ -235,26 +360,20 @@ function peek(from::GenericIOBuffer, T::Union{Type{Int16},Type{UInt16},Type{Int3
     return x
 end
 
-function read(from::GenericIOBuffer, T::Union{Type{Int16},Type{UInt16},Type{Int32},Type{UInt32},Type{Int64},Type{UInt64},Type{Int128},Type{UInt128},Type{Float16},Type{Float32},Type{Float64}})
+function read(from::GenericIOBuffer, T::MultiByteBitNumberType)
     x = peek(from, T)
     from.ptr += sizeof(T)
     return x
 end
 
-function read_sub(from::GenericIOBuffer, a::AbstractArray{T}, offs, nel) where T
+function read_sub(from::GenericIOBuffer, a::MutableDenseArrayType{T}, offs, nel) where T
```

Comment:
This whole function should be removed. It is only used in `readbytes!` and there it should be replaced with `copyto!` like what I do in https://github.com/JuliaIO/InputBuffers.jl/blob/da9b2315f11cd041a247bb2a088d6e1f28ccd9a4/src/InputBuffers.jl#L79-L88